### PR TITLE
Add FXIOS-4683 [v105] first run onboarding update

### PR DESCRIPTION
**Pull request by yoanarios**
Issue #11484 
[Jira](https://mozilla-hub.atlassian.net/browse/FXIOS-4683)

- Add logic to viewModel to support to versions of onboarding
- Remove wallpaper card based on upgrade onboarding feature flag
- Update text for title and description
- Add background illustration just for MR version
- Includes UX review fixes for Update onboarding because it will be needed for this story
- Use async hasAccount to show Update onboarding to show the current number of cards

### CODE REVIEW

File: Client/Frontend/Intro/IntroViewController.swift [hunk 1]
```diff
@@ -6,10 +6,13 @@ import Foundation
...
 import Shared
 
-class IntroViewController: UIViewController {
+class IntroViewController: UIViewController, OnboardingViewControllerProtocol {
+
     private var viewModel: IntroViewModel
     private let profile: Profile
     private var onboardingCards = [OnboardingCardViewController]()
+    var notificationCenter: NotificationProtocol = NotificationCenter.default
+    var didFinishClosure: (() -> Void)?
```
Comment (thatswinnie) [inline Client/Frontend/Intro/IntroViewController.swift:15]: Wondering if this should be renamed as it's not clear what finished without looking at the code 🤔

File: Client/Frontend/Intro/IntroViewModel.swift [hunk 2]
```diff
@@ -44,37 +44,51 @@ struct IntroViewModel: OnboardingViewModelProtocol, FeatureFlaggable {
         }
     }
 
-    var enabledCards: [InformationCards] {
-        guard let wallpaperVersion: WallpaperVersion = featureFlags.getCustomState(for: .wallpaperVersion) else {
-            return [.welcome, .wallpapers, .signSync]
-        }
+    var isMROnboardingVersion: Bool {
```
Comment (thatswinnie) [inline Client/Frontend/Intro/IntroViewModel.swift:47]: Can we rename this to be more specific? And additionally create a task to remove this once the MR is released? Don't want to end up at the next MR and still have this in place and not knowing with MR it was for.

File: Client/Frontend/Browser/BrowserViewController.swift [hunk 3]
```diff
@@ -2216,33 +2216,39 @@ extension BrowserViewController {
...
+        present(updateViewController, animated: animated) {
+            self.setupHomepageOnBackground()
         }
     }
 
     private func showProperIntroVC() {
         let introViewModel = IntroViewModel()
         let introViewController = IntroViewController(viewModel: introViewModel, profile: profile)
-        introViewController.didFinishClosure = { controller, _ in
+        introViewController.didFinishFlow = {
```
Comment (electricRGB) [inline Client/Frontend/Browser/BrowserViewController.swift:2249]: `didFinishFlow` reeks of Controller pattern and I love the sneaking in of it!

File: Client/Frontend/Intro/IntroViewModel.swift (hunk 2 above)
Comment (yoanarios) [reply Client/Frontend/Intro/IntroViewModel.swift]: Update name and I've updated the acceptance criteria of a future task to remove this logic

File: Client/Frontend/Update/UpdateViewController.swift [hunk 4]
```diff
@@ -70,10 +62,17 @@ class UpdateViewController: UIViewController, OnboardingViewControllerProtocol {
...
+    deinit {
+        notificationCenter.removeObserver(self)
+    }
+
     override func viewDidLoad() {
         super.viewDidLoad()
 
         setupView()
+        setupNotifications(forObserver: self,
+                                   observing: [.DisplayThemeChanged])
```
Comment (electricRGB) [inline Client/Frontend/Update/UpdateViewController.swift:74]: Intenting is weird.

**Review by electricRGB — APPROVED** [review]

**Review by thatswinnie — APPROVED** [review]

**Comment by yoanarios** [comment]
Bitrise is green
Screen Shot 2022-08-22 at 12 57 37 PM

**Pull request merged**